src/coreclr/gc/gc.cpp

```diff
@@ -28134,7 +28134,7 @@ void gc_heap::thread_final_regions (bool compact_p)
         heap_segment* current_region = heap_segment_rw (generation_start_segment (generation_of (gen_idx)));
         dprintf (REGIONS_LOG, ("gen%d start from %Ix", gen_idx, heap_segment_mem (current_region)));
 
-        while (current_region = find_first_valid_region (current_region, compact_p))
+        while ((current_region = find_first_valid_region (current_region, compact_p)))
```

Comment:
this change is not needed anymore?

Reply:
other than this it LGTM!

Reply:
its needed to avoid a build break on linux:

```
  In file included from /home/magodse/fork/runtime/src/coreclr/gc/gcwks.cpp:29:
  /home/magodse/fork/runtime/src/coreclr/gc/gc.cpp: In static member function ‘static void WKS::gc_heap::thread_final_regions(bool)’:
  /home/magodse/fork/runtime/src/coreclr/gc/gc.cpp:28137:31: error: suggest parentheses around assignment used as truth value [-Werror=parentheses]
  28137 |         while (current_region = find_first_valid_region (current_region, compact_p))
        |                ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```